dkms

```diff
@@ -169,6 +169,14 @@ set_module_suffix()
     kernel_test="${1-:$(uname -r)}"
     module_suffix=".ko"
     [[ $(VER $kernel_test) < $(VER 2.5) ]] && module_suffix=".o"
+    grep -q '\.gz:' /lib/modules/$kernel_test/modules.dep 2>/dev/null && gz_modules=1
```

Comment:
Why not checking the kernel config for MODULE_COMPRESS_XZ / MODULE_COMPRESS_GZIP ?